cert-manager/ingress-rules-syncer/controllers/ingress_controller.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	syncFromLabelKey = "ingress-rules-syncer.x-k8s.io/sync-from"
+)
+
+// IngressReconciler reconciles a Ingress object
+type IngressReconciler struct {
+	client.Client
+	Log logr.Logger
+}
+
+//+kubebuilder:rbac:groups=networking.k8s.io,resources=ingresses,verbs=get;list;watch;update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// If an Ingress has a label 'ingress-rules-syncer.x-k8s.io/sync-from', the value of that
+// label will be used as the name of the Ingress resource to source 'rules' from.
+// These will be copied from the referenced/named Ingress resource.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.7.0/pkg/reconcile
+func (r *IngressReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("ingress", req.NamespacedName)
+
+	var ing networkingv1beta1.Ingress
+	if err := r.Client.Get(ctx, req.NamespacedName, &ing); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// Base case where an Ingress does NOT have a 'sync-from' label
+	if ing.Labels == nil || ing.Labels[syncFromLabelKey] == "" {
```

Comment:
I guess you don't need to verify if Labels is nil. 

When you call an empty Map, with an unexistant value it will work. -> https://play.golang.org/p/0eLKBSSyyZJ